File: Main/Backup/Formularios/frmInputGeneral.vb
Imports System.Windows.Forms

Public Class frmInputGeneral

   Private sInputGeneral As String

   Public ReadOnly Property ResultadoInput() As String
      Get
         Return sInputGeneral
      End Get
   End Property

   Private Sub OK_Button_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles OK_Button.Click

      If txtResultado.Text.Trim <> "" Then
         sInputGeneral = txtResultado.Text
         INPUT_GENERAL = txtResultado.Text
         Me.DialogResult = System.Windows.Forms.DialogResult.OK
         Me.Close()
      Else
         MensajeError("Dato no válido")
         txtResultado.Focus()
      End If

   End Sub

   Private Sub Cancel_Button_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Cancel_Button.Click
      INPUT_GENERAL = ""
      Me.DialogResult = System.Windows.Forms.DialogResult.Cancel
      Me.Close()
   End Sub

   Public Sub PasarInfoVentana(ByVal sCaption As String, _
                               ByVal sLabel As String)

      Me.Text = sCaption
      lblTip.Text = sLabel

   End Sub

End Class

File: Prex.Utils/Prex.Utils/Misc/Forms/FrmInputGeneral.cs
using System;
using System.Windows.Forms;

namespace Prex.Utils.Misc.Forms
{
    public partial class FrmInputGeneral : Form
    {

        private string sInputGeneral;
        public string ResultadoInput => sInputGeneral;

        public FrmInputGeneral()
        {
            InitializeComponent();
        }

        private void OK_Button_Click(object sender, EventArgs e)
        {
            if (txtResultado.Text.Trim() != "")
            {
                sInputGeneral = txtResultado.Text;
                Configuration.PrexConfig.INPUT_GENERAL = txtResultado.Text;
                this.DialogResult = System.Windows.Forms.DialogResult.OK;
                this.Close();
            }
            else
            {
                Prex.Utils.MensajesForms.MostrarError("Dato no válido");
                txtResultado.Focus();
            }

        }

        private void Cancel_Button_Click(object sender, EventArgs e)
        {
            Configuration.PrexConfig.INPUT_GENERAL = "";
            DialogResult = System.Windows.Forms.DialogResult.Cancel;
            Close();
        }

        public void PasarInfoVentana(string titulo, string label)
        {
            Text = titulo;
            lblTip.Text = label;
        }
    }
}
